src/date-input/index.tsx

```diff
@@ -0,0 +1,80 @@
+import { create, tsx } from '@dojo/framework/core/vdom';
+import icache from '@dojo/framework/core/middleware/icache';
+import theme from '../middleware/theme';
+import Calendar from '../calendar';
+import TextInput from '../text-input';
+import Icon from '@dojo/widgets/icon';
+import Popup from '../popup';
+import * as css from '../theme/default/date-input.m.css';
+
+export interface DateInputProperties {
+	/** Set the latest date the calendar will display (it will show the whole month but not allow later selections) */
+	maxDate?: Date;
+	/** Set the earliest date the calendar will display (it will show the whole month but not allow previous selections) */
+	minDate?: Date;
+	/** Callback fired when the input value changes */
+	onValue?(date: Date): void;
+	/** The current value */
+	value?: Date;
```

Comment:
we've been making these widgets less controlled as of late, to follow this pattern, this should be `initialValue`